hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -152,6 +154,22 @@ private static void setupHoodieKeyOptions(Configuration conf, CatalogTable table
     }
   }
 
+  /**
+   * Sets up the cleaning options from the table definition.
+   */
+  private static void setupCleaningOptions(Configuration conf) {
+    int commitsToRetain = conf.getInteger(FlinkOptions.CLEAN_RETAIN_COMMITS);
+    int minCommitsToKeep = conf.getInteger(FlinkOptions.ARCHIVE_MIN_COMMITS);
+    if (commitsToRetain >= minCommitsToKeep) {
```

Comment:
I don't quietly understand why not align with the `HoodieCompactionConfig#build` to check `minCommitsToKeep` and `commitsToRetain`, and throw exception if users set wrong value.

Reply:
For flink streaming read, we need to keep enough logs files for reading. And, i don't want to expose too much options for sql users.

Reply:
@danny0405 so here we would change `FlinkOptions.CLEAN_RETAIN_COMMITS`, `FlinkOptions.ARCHIVE_MIN_COMMITS`, `ARCHIVE_MIN_COMMITS ` to a larger default value to remove the logic here?

Reply:
Yes, Flink user can set up whatever retain commits they like.